priam/src/main/java/com/netflix/priam/services/SnapshotMetaService.java

```diff
@@ -0,0 +1,227 @@
+            return;
+        }
+
+        try {
+            Instant snapshotInstant = DateUtil.getInstant();
+            snapshotName = generateSnapshotName(snapshotInstant);
+            logger.info("Initializaing SnapshotMetaService for taking a snapshot {}" + snapshotName);
+
+            //Perform a cleanup of old snapshot meta_v2.json files, if any, as we don't want our disk to be filled by them.
+            //These files may be leftover
+            // 1) when Priam shutdown in middle of this service and may not be full JSON
+            // 2) No permission to upload to backup file system.
+            metaFileManager.cleanupOldMetaFiles();
+
+            //TODO: enqueue all the old backup folder for upload/delete, if any, as we don't want our disk to be filled by them.
+            //processOldSnapshotV2Folders();
+
+            //Take a new snapshot
+            cassandraOperations.takeSnapshot(snapshotName);
+
+            //Process the snapshot and upload the meta file.
+            processSnapshot(snapshotInstant).uploadMetaFile(true);
+
+            logger.info("Finished processing snapshot meta service");
+        } catch (Exception e) {
+            logger.error("Error while executing SnapshotMetaService", e);
+        }
+
+    }
+
+    MetaFileWriterBuilder.UploadStep processSnapshot(Instant snapshotInstant) throws Exception {
+        dataStep = metaFileWriter.newBuilder().startMetaFileGeneration(snapshotInstant);
+        initiateBackup(SNAPSHOT_FOLDER, backupRestoreUtil);
+        return dataStep.endMetaFileGeneration();
+    }
+
+    private File getValidSnapshot(File snapshotDir, String snapshotName) {
+        for (File fileName : snapshotDir.listFiles())
+            if (fileName.exists() && fileName.isDirectory() && fileName.getName().matches(snapshotName))
+                return fileName;
+        return null;
+    }
+
+    @Override
+    public String getName() {
+        return JOBNAME;
+    }
+
+
+    private ColumnfamilyResult convertToColumnFamilyResult(String keyspace, String columnFamilyName, Map<String, List<FileUploadResult>> filePrefixToFileMap) {
+        ColumnfamilyResult columnfamilyResult = new ColumnfamilyResult(keyspace, columnFamilyName);
+        filePrefixToFileMap.entrySet().forEach(sstableEntry -> {
+            ColumnfamilyResult.SSTableResult ssTableResult = new ColumnfamilyResult.SSTableResult();
+            ssTableResult.setPrefix(sstableEntry.getKey());
+            ssTableResult.setSstableComponents(sstableEntry.getValue());
+            columnfamilyResult.addSstable(ssTableResult);
+        });
+        return columnfamilyResult;
+    }
+
+    @Override
+    protected void processColumnFamily(final String keyspace, final String columnFamily, final File backupDir) throws Exception {
+        File snapshotDir = getValidSnapshot(backupDir, snapshotName);
+        // Process this snapshot folder for the given columnFamily
+        if (snapshotDir == null) {
```

Comment:
Short answer: no, this method will ensure that all active CF are accounted for. 
Long version: 
Before we execute this command, we have asked C* to take a snapshot which means a valid snapshot for all the "active" CF should exist. 
We walk over folder structure in C* data folder and there could be "folders" in data folder which are not really CF or some leftover from previous, user errors etc.  So we really need this to avoid such bad case. 
Also, this method is per CF data folder so if it really is not an active CF, we skipping that CF is valid.